openhgnn/models/MAGNN.py

```diff
@@ -591,3 +598,23 @@ def svm_test(X, y, test_sizes=(0.2, 0.4, 0.6, 0.8), repeat=10):
         result_macro_f1_list.append((np.mean(macro_f1_list), np.std(macro_f1_list)))
         result_micro_f1_list.append((np.mean(micro_f1_list), np.std(micro_f1_list)))
     return result_macro_f1_list, result_micro_f1_list
+
+
+def metapath_generator(ntypes,etypes,length,num):
+    ntypes_num = len(ntypes)
+    metapath_list = []
+    for i in range(num):
+        temp = ""
+        random_index = random.randrange(ntypes_num)
+        temp = temp + ntypes[random_index]
+        for j in range(length-1):
+            temp = temp + '-'
+            random_index = random.randrange(ntypes_num)
+            temp_e = temp[-2] + '-' + ntypes[random_index]
+            while temp_e not in etypes:
+                random_index = random.randrange(ntypes_num)
+                temp_e = temp[-2] + '-' + ntypes[random_index]
+            temp = temp + ntypes[random_index]
+        if temp not in metapath_list:
+            metapath_list.append(temp)
+    return metapath_list
```

Comment:
MAGNN model request all metapath in metapath_list to be symmetric.
This function seems unable to do that...